### PR TITLE
Update setup.py

### LINKED ISSUE #4: Data Preprocessing
- [x] Determine format of input data

- [x] Create Data loaders

- [x]  Create Config file for input parameters

- [x] Code  Data Preprocessing

- [x] Determine structure of masks

- Mask will be binary images (0,1)
- When data is sampled for each task, multiply masks by an integer from 1 to num_tasks.
- The mask can be used as it is if the loss function can perform cross entropy loss with sparse inputs. 

- Else write function to convert a mask to such that its depth dimension are size (num_tasks) and pixels are one-hot-encoded along the depth dimension.

### PR DESCRIPTION
**Pull request by Temiloluwa**
bugfix to find packages call

### CODE REVIEW

**Comment by Temiloluwa** [comment]
Created this at the beginning to describe our workflow. Made many changes past this

**Comment by Temiloluwa** [comment]
Completed the data preprocessing step.

-  Created config file
- Created a structure for our data format
- For each training episode:
    - MetaTrain, MetaVal or MetaTest Data can be loaded using the dataloader

File: leo_segmentation/run.py [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+import argparse
 import torch
 
-def train_model(saved_model_path:str):
-    print(f"I am training a model saved at {saved_model_path}")
-    return
+parser = argparse.ArgumentParser(description='Specify train or inference dataset')
+parser.add_argument("-d", "--dataset", type=str, nargs=1, default="pascal_voc")
+args = parser.parse_args()
+dataset = args.dataset[0]
```
Comment (GopikaSudhakaran) [inline leo_segmentation/run.py:12]: dataset = args.dataset #dataset[0] is returning first letter

File: leo_segmentation/data.py [hunk 2]
```diff
@@ -1 +1,137 @@
...
+                                        for image_path in class_paths]
+                                        for class_paths in path_array])
+            if i == 0:
+                batch_embeddings = np.empty((self.config["num_tasks"],) + embedding_array.shape)
+                batch_masks = np.empty((self.config["num_tasks"],) + mask_array.shape)
+            batch_embeddings[i] = embedding_array
+            batch_masks[i] = mask_array
+        
+        tr_data = batch_embeddings[:,:,:tr_size,:,:]
+        tr_data_masks = batch_masks[:,:,tr_size:,:,:]
```
Comment (GopikaSudhakaran) [inline leo_segmentation/data.py]: tr_data_masks = batch_masks[:, :, :tr_size, :, :] #current logic is resulting in  val_size

File: leo_segmentation/data.py [hunk 3]
```diff
@@ -1 +1,137 @@
...
+        val_data = batch_embeddings[:,:,:tr_size,:,:]
```
Comment (GopikaSudhakaran) [inline leo_segmentation/data.py]: val_data = batch_masks[:, :, tr_size:, :, :] #current logic is resulting in  tr_size

File: leo_segmentation/data.py [hunk 4]
```diff
@@ -1 +1,137 @@
...
+            image_paths = []
+            for _, class_name in enumerate(shuffled_list):
+                all_images = _all_class_images[class_name]
+                all_images = np.random.choice(all_images, sample_count, replace=False)
+                error_message = f"{len(all_images)} == {sample_count} failed"
+                assert len(all_images) == sample_count, error_message
+                image_paths.append(all_images)
+            
+            path_array = np.array(image_paths)
+            embedding_array = np.array([[_image_mask_embeddings[image_path][0]
```
Comment (GopikaSudhakaran) [inline leo_segmentation/data.py:103]: each embedding 2D (32, 20) so ideally this embedding array of batch_size (input_task) = 12, tr + val = 8, should result in (12,8,32,20) but the above logic is resulting in (12, 8, 8, 32, 20) am I miss understanding something here?

Comment (Temiloluwa) [reply leo_segmentation/data.py:103]: _Good observation_ . In the doc-string I wrote this:
`
Returns:
            A tuple of pytorch tensors:
            DIMS = Image or Mask shape
            - tr_data: (batch_size, num_classes, tr_size, DIMS): training image embeddings
            - tr_masks: (batch_size, num_classes, tr_size, DIMS): training image masks
            - val_data: (batch_size, num_classes, val_size, DIMS): validation image embeddings
            - val_masks: (batch_size, num_classes, val_size, DIMS): validation image masks 
  `

Please confirm it does that. Since we are going to train per class, there is also a dimension for class. We were discussing about averaging codes per class for the relational network.

File: leo_segmentation/data.py (hunk 3 above)
Comment (Temiloluwa) [reply leo_segmentation/data.py]: Bug fixed.
saw another bug with tr_data_masks.

File: leo_segmentation/run.py (hunk 1 above)
Comment (Temiloluwa) [reply leo_segmentation/run.py:12]: image
In the command line or IDE it should be ran like this

image

Basically only the dataset name. check if you are getting the same error

File: leo_segmentation/data.py (hunk 4 above)
Comment (GopikaSudhakaran) [reply leo_segmentation/data.py:103]: I get your point now because both the dimension Num_class (8) and tr + val (8) i got confused

**Review by GopikaSudhakaran — APPROVED** [review]
